components/raftstore/src/store/snap.rs

```diff
@@ -2046,6 +2047,7 @@ impl TabletSnapManager {
             ));
         }
         file_system::clean_up_trash(&self.base)?;
+        file_system::clean_up_dir(&self.base, SNAP_GEN_PREFIX)?;
```

Comment:
Can `init` be deleted as `new` does the same thing as `init`?

Reply:
But some caller using `new_without_init` to create `TabletSnapshotManager`.

Reply:
Like what case?

Reply:
I don't see why it's not initialized here. It's almost initialized immediately after being built.

Reply:
The reason is that some called maybe using `""` to create `SnapshotManagerCore`. I think `TabletSnapManager` should redefine `new` and `init`, I think they do the same things .

Reply:
`new_without_init` is introduced yesterday, how come those test cases pass in the past?